src/odfesqlodbc/es_statement.cpp

```diff
@@ -148,6 +152,29 @@ RETCODE ExecuteStatement(StatementClass *stmt, BOOL commit) {
     return CleanUp();
 }
 
+SQLRETURN GetMoreResults(StatementClass *stmt) {
+    if (stmt == NULL)
+        return SQL_ERROR;
+    ConnectionClass *conn = SC_get_conn(stmt);
+    QResultClass *q_res = SC_get_Result(stmt);
+    if ((q_res == NULL) && (conn == NULL)) {
+        return SQL_ERROR;
+    }
+    int get_more_result = SQL_ERROR;
+    do {
+        ESResult *es_res = ESGetResult(conn->esconn);
+        if (es_res != NULL) {
+            get_more_result = SQL_SUCCESS;
+            schema_type *doc_schema = ESGetDocSchema(conn);
+            CC_Assign_Table_Data(es_res->es_result_doc, q_res, *doc_schema,
+                                 *(q_res->fields));
+        } else {
+            get_more_result = SQL_ERROR;
+        }
+    } while (get_more_result == SQL_SUCCESS);
```

Comment:
We don't need this flag, you can instead make this an infinite loop where you break if es_res == NULL.